File: 1122_relative_sort_array/relativeSortArray.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        count = [0] * 1001

        for i in arr1:
            count[i] += 1

        pos = 0

        for i in arr2:
            while count[i] > 0:
                arr1[pos] = i
                pos += 1
                count[i] -= 1

        for i in range(len(count)):
            while count[i] > 0:
                arr1[pos] = i
                pos += 1
                count[i] -= 1

        return arr1

File: 1122_relative_sort_array/relativeSortArray.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] count = new int[1001];

        for (int i : arr1) {
            count[i]++;

        }

        int pos = 0;

        for (int i : arr2) {
            while (count[i] > 0) {
                count[i]--;
                arr1[pos] = i;
                pos++;
            }
        }

        for (int i = 0; i < count.length; i++) {
            while (count[i] > 0) {
                count[i]--;
                arr1[pos] = i;
                pos++;
            }
        }

        return arr1;
    }
}